src/bin/firebase-functions.ts

```diff
@@ -15,20 +15,24 @@ Arguments:
   process.exit(1);
 }
 
+let functionsDir = '.';
+
 const args = process.argv.slice(2);
-if (args.length != 1) {
-  printUsage();
+if (args.length > 1) {
+  if (args[0] === '-h' || args[0] === '--help') {
+    printUsageAndExit();
+  }
+  functionsDir = args[0];
 }
-const functionsDir = args[0];
 
 let server;
 const app = express();
 
 app.get('/stack.yaml', async (req, res) => {
```

Comment:
we can also add `app.post('__/quitquitquit')`

The prototypes used this instead of pkill because some of the launchers don't kill their children when killed (e.g. `go run` will leave the server running, but `POST http://localhost:$PORT/__/quitquitquit` will tell the service to shut down and the runner will gracefully follow.